teachertool/src/hooks/useJsonDocRequest.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from "react";
+import { RequestStatus } from "../types";
+import { fetchJsonDocAsync } from "../services/backendRequests";
+
+export function useJsonDocRequest<T>(
+    url: string,
+    statusCb: (status: RequestStatus) => void,
```

Comment:
Is "Cb" callback? If so, nit request to simply spell it out. Took me a little while to figure out what it was.